### PR TITLE
Modify List feature to disallow parameters

### PR DESCRIPTION
**Pull request by Jweewee**
Enabled stricter usage of List feature by ensuring it does not allow any parameters to keep formatting as simple and strict as possible.

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/ListCommandParser.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+     * Parses the given {@code String} of arguments in the context of the ListCommand
+     * and returns a ListCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public ListCommand parse(String args) throws ParseException {
+        if (args.isEmpty()) {
+            return new ListCommand();
+        } else {
+            throw new ParseException(
+                String.format(ListCommand.MESSAGE_USAGE));
```
Comment (yucongkoo) [inline src/main/java/seedu/address/logic/parser/ListCommandParser.java:21]: Just wondering, why are you using String.format() on a pure string, would using
 ```new ParseException(ListCommand.MESSAGE_USAGE)``` 
directly be more straightforward here?

File: src/test/java/seedu/address/logic/parser/AddressBookParserTest.java [hunk 2]
```diff
@@ -85,7 +85,8 @@ public void parseCommand_help() throws Exception {
     @Test
     public void parseCommand_list() throws Exception {
         assertTrue(parser.parseCommand(ListCommand.COMMAND_WORD) instanceof ListCommand);
-        assertTrue(parser.parseCommand(ListCommand.COMMAND_WORD + " 3") instanceof ListCommand);
+        assertThrows(ParseException.class, String.format(ListCommand.MESSAGE_USAGE), ()
```
Comment (yucongkoo) [inline src/test/java/seedu/address/logic/parser/AddressBookParserTest.java:88]: Same question as stated above regarding usage of redundant String.format()

Comment (Jweewee) [reply src/test/java/seedu/address/logic/parser/AddressBookParserTest.java]: You are right, have pushed the changes to make code more concise.

**Review by yucongkoo — APPROVED** [review]
LGTM

**Pull request merged**